src/server/worker/common/common.go

```diff
@@ -49,6 +50,29 @@ func HashDatum(pipelineName string, pipelineSalt string, inputs []*Input) string
 	return client.DatumTagPrefix(pipelineSalt) + hex.EncodeToString(hash.Sum(nil))
 }
 
+// DatumIDV2 computes the ID of a datum.
+// TODO: This needs more discussion.
+func DatumIDV2(inputs []*Input) string {
```

Comment:
Why is `/` replaced with`_` is that so we can write it to object storage?  Does files need to be sorted?

Reply:
The datum id is used as a directory name in the stats commit. The "This needs more discussion" mainly refers to this as well as the fact that the datum id is a concatenation of the constituent input file paths (with the `/` -> `_`) separated by dashes. I am definitely not committed to this representation, but this is usable for now. The datum ids need to map back to the input files and be sortable for the streaming operations. (@jdoliner)

Reply:
So it's important that we use a concat of the filenames as the datum id so that when we merge the output the order of the merge is determined by the lexicographic order of the datums. I'm still unclear on why we need to do the `/` -> `_` mapping. It seems like that could potentially create a rare bug the case where you have `_`s in filenames. For example the files `foo_bar` and `foo/bar` can be separate files in the same job.

Reply:
The datum directory name in the stats commit is the datum id. If the datum id has slashes in it, then we cannot use it as the directory name (the slashes would be interpreted as file separators). So, the `/` -> `_` mapping gets around this for the time being. A different character would be fine, as long as there is no `/` in the datum id.